srml/staking/src/phragmen.rs

```diff
@@ -92,15 +92,14 @@ pub struct Edge<AccountId> {
 ///
 /// Returns an Option of elected candidates, if election is performed.
 /// Returns None if not enough candidates exist.
-pub fn elect<T: Trait + 'static, FR, FN, FV, FS>(
-		get_rounds: FR,
-		get_validators: FV,
-		get_nominators: FN,
-		stash_of: FS,
-		minimum_validator_count: usize,
-		config: ElectionConfig<BalanceOf<T>>,
+pub fn elect<T: Trait + 'static, FN, FV, FS>(
+	validator_count: usize,
```

Comment:
rename `rounds` to `validator_count` - to minimise use of jargon.

also wtf is this passing in a closure only to call it immediately?

Reply:
See the open discussions here. https://github.com/paritytech/substrate/pull/1915#discussion_r263425454

I would personally not do that and pre-process the data in `lib.re` and pass everything in as a vectors.